libminifi/include/Exception.h

```diff
@@ -44,49 +43,56 @@ enum ExceptionType {
   MAX_EXCEPTION
 };
 
-// Exception String 
 static const char *ExceptionStr[MAX_EXCEPTION] = { "File Operation", "Flow File Operation", "Processor Operation", "Process Session Operation", "Process Schedule Operation", "Site2Site Protocol",
     "General Operation", "Regex Operation" };
 
-// Exception Type to String 
 inline const char *ExceptionTypeToString(ExceptionType type) {
   if (type < MAX_EXCEPTION)
     return ExceptionStr[type];
   else
     return NULL;
 }
 
-// Exception Class
-class Exception : public std::exception {
- public:
-  // Constructor
-  /*!
-   * Create a new exception
-   */
-  Exception(ExceptionType type, std::string errorMsg)
-      : _type(type),
-        _errorMsg(std::move(errorMsg)) {
-  }
+namespace detail {
+inline size_t StringLength(const char* str) { return strlen(str); }
 
-  // Destructor
-  virtual ~Exception() noexcept {
-  }
-  virtual const char * what() const noexcept {
+template<size_t L>
+constexpr size_t StringLength(const char (&str)[L]) { return L; }
 
-    _whatStr = ExceptionTypeToString(_type);
+inline size_t StringLength(const std::string& str) { return str.size(); }
 
-    _whatStr += ":" + _errorMsg;
-    return _whatStr.c_str();
-  }
+template<typename... SizeT>
+size_t sum(SizeT... ns) {
+  size_t result = 0;
+  (void)(std::initializer_list<size_t>{(
+      result += ns
+      )...});
+  return result; // (ns + ...)
+}
 
- private:
-  // Exception type
-  ExceptionType _type;
-  // Exception detailed information
-  std::string _errorMsg;
-  // Hold the what result
-  mutable std::string _whatStr;
+template<typename... Strs>
+std::string StringJoin(Strs&&... strs) {
```

Comment:
I think this should be moved to our StringUtils if we haven't had this feature yet.

Reply:
Good point, we do have a join there.

Reply:
moved to StringUtils in dbc0033